modules/ui/src/main/helper/HtmlHelper.scala

```diff
@@ -24,10 +26,16 @@ object HtmlHelper:
   def copyMeInput(content: String): Tag =
     copyMe(input(spellcheck := "false", readonly, value := content))
 
-  private def copyMe(target: Tag): Tag =
+  def copyMePgnLink(url: String, name: Frag, distinct_copy_val: Future[String]): Future[Tag] =
+    distinct_copy_val.map(res => copyMe(a(targetBlank, href := url)(name), Some(res)))
+
+  private def copyMe(target: Tag, button_value: Option[String] = None): Tag =
+    val base_button = button(cls := "copy-me__button button button-metal", dataIcon := Icon.Clipboard)
     div(cls := "copy-me")(
       target(cls := "copy-me__target"),
-      button(cls := "copy-me__button button button-metal", dataIcon := Icon.Clipboard)
+      button_value match
+        case Some(b) => base_button(value := button_value)
```

Comment:
You're not using the `b` you just extracted. The whole `match` is indeed superfluous and `baseButton(value := buttonValue)` would have be enough